x-pack/plugins/canvas/public/lib/aeroelastic/types.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+type f64 = number; // eventual AssemblyScript compatibility; doesn't hurt with vanilla TS either
+type f = f64; // shorthand
+
+export type vector2d = [f, f, f] & ReadonlyArray<f> & { __nominal: 'vector2d' };
+export type vector3d = [f, f, f, f] & ReadonlyArray<f> & { __nominal: 'vector3d' };
+
+export type transformMatrix2d = [f, f, f, f, f, f, f, f, f] &
+  ReadonlyArray<f> & { __nominal: 'transformMatrix2d' };
+export type transformMatrix3d = [f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, f] &
+  ReadonlyArray<f> & { __nominal: 'transformMatrix3d' };
+
+export interface Meta {
+  silent: boolean;
+}
+export type ActionId = number;
+export type TypeName = string;
+export type NodeResult = any;
```

Comment:
What values do we expect to be type `NodeResult`? Is there anyway we can validate this at all? Same with `payload`.

Reply:
Good point, those are just placeholders at the moment to pass the linter, I'd specialize them in the next round. For example, the actions would be of fixed shape. This PR isn't attempting to do it yet.

Reply:
Re the suggestion for review by someone more experienced: yes I'd stand to learn from improvement ideas as I'm a TS rookie, though our team has few developers. @rayafratkina @clintandrewhall what can we do in this case, should someone from another team jump in? It's been open for about two weeks, and there are some functional improvements to be done (eg. the Z-order rework that's blocking one of Catherine's PRs), and there's a chance that any time, some urgent fix is needed, in which case it'll have to be implemented twice. 

I'm also planning more TypeScript changes - which could be somewhat less trivial than these so far -, but it's not ideal to do without this merged too.

In short, these kinds of code-altering, yet non-functional rework items are blocking downstream work on the same codebase until merged.

An alternative to asking another team to review: As this PR looks like an overall improvement plus partial TypeScript use, maybe the criterion can be, 1) does it improve the codebase, 2) do we want to get on with other things, 3) can we address what would be additional PR feedback on this with a subsequent PR.